### PR TITLE
PUBDEV-6290: Fixing AutoML leader slot in R for single model

### PR DESCRIPTION
**Pull request by ledell**
- Fixed a bug on the AutoML R client described in [PUBDEV-6290](https://0xdata.atlassian.net/projects/PUBDEV/issues/PUBDEV-6290)
- Removed the single NA row for empty leaderboards in R
- Added warning for when there's an empty leaderboard
- The `h2o.getAutoML()` function was also not working for empty leaderboards in R (now fixed)
- Fixed unit test for empty leaderboard (was checking for 1 row, now checks for 0 rows)
- There are a lines changed because of deleted spaces (must have been introduced by my editor...)

### CODE REVIEW

**Comment by ledell** [comment]
Can test it out using this code here (doesn't really need to be a unit test though):

```
library(h2o)
h2o.init()

# Import a sample binary outcome train/test set into H2O
train <- h2o.importFile("http://s3.amazonaws.com/erin-data/higgs/higgs_train_10k.csv")
y <- "response"
x <- setdiff(names(train), y)
train[,y] <- as.factor(train[,y])

aml <- h2o.automl(x = x, y = y,
                  training_frame = train,
                  max_runtime_secs = 3,
                  seed = 1)
```
Now we have a zero row leaderboard (just like Python), and `h2o.getAutoML()` gives an error message now:
```
> aml@leaderboard
[1] model_id             auc                  logloss              mean_per_class_error rmse                
[6] mse                 
<0 rows> (or 0-length row.names)

[0 row x 6 columns] 
> aa <- h2o.getAutoML(aml@project_name)
Warning message:
In h2o.getAutoML(aml@project_name) :
  The leaderboard contains zero models: try running AutoML for longer (the default is 1 hour).
```

File: h2o-r/h2o-package/R/automl.R [hunk 1]
```diff
@@ -266,15 +267,20 @@ h2o.automl <- function(x, y, training_frame,
...
+    leader <- h2o.getModel(automl_job$leaderboard$models[[1]]$name)
   } else {
-      # create a phony leader
-      Class <- paste0("H2OBinomialModel")
-      leader <- .newH2OModel(Class = Class,
-                             model_id = "dummy")
+    # create a phony leader
+    Class <- paste0("H2OBinomialModel")
+    leader <- .newH2OModel(Class = Class,
+                           model_id = "dummy")
```
Comment (seb-h2o) [inline h2o-r/h2o-package/R/automl.R]: what prevents you from reusing `h2o.getAutoML` directly from L254? 
Would be better than having to duplicate code:
```
aml <- h2o.getAutoML(res$job$dest$name)
aml$project_name <- build_control$project_name
return(aml)
```

**Comment by ledell** [comment]
@seb-h2o You're right, I should just wrap the code to retrieve and process the leaderboard and leader in function and call it twice.

File: h2o-r/h2o-package/R/automl.R (hunk 1 above)
Comment (ledell) [reply h2o-r/h2o-package/R/automl.R]: @seb-h2o Yeah you're right, I can just do that.  Thanks!

Comment (ledell) [reply h2o-r/h2o-package/R/automl.R]: @seb-h2o I changed the `h2o.automl()` function to make use of `h2o.getAutoML()`. Does everything else look good?

**Pull request merged**